gui/middleware/zfs.py

```diff
@@ -59,6 +59,13 @@ def _vdev_type(name):
     return False
 
 
+def promote_zfs(name):
+    datasets = list(libzfs.ZFS().datasets)
+    for dataset in datasets:
```

Comment:
Iterating over all datasets is not good. ZFS object has a `get_dataset` method.

Reply:
@william-gr Yeah, that sounds better.